### PR TITLE
Add input examples tutorial

### PR DESCRIPTION
**Pull request by cbscribe**
Adds a tutorial about InputEvent* types, with example usage.

### CODE REVIEW

File: tutorials/inputs/input_examples.rst [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+.. tabs::
+ .. code-tab:: gdscript GDScript
+
+    # input event - runs when the input happens
+    func _input(event):
+        if event.is_action_pressed("jump"):
+            jump()
+
+    # polling - runs every frame
+    func _process(delta):
```
Comment (TheDuriel) [inline tutorials/inputs/input_examples.rst]:
```suggestion
    func _physics_process(delta):
```

_process runs at a variable rate, independent from framerate. (You can get 20 process ticks at 60fps. With _physics_process still ticking at 60.) If this happens, the original code would introduce stutter.

All motion should occur in _physics_process. Let's not show newbies code that has the potential of breaking down later.

**Comment by TheDuriel** [comment]
Another thing i would consider adding, (great overview btw.) is how to deal with button combinations. Be it for modifier keys, or multi button combinations on the gamepad. This is easy to do in a _physics_process, but it is _also_ possible with _input through use of the Input singleton. It feels a bit dirty, but it is easier then keeping a list of pressed keys you check every frame to see if there is a combination.

```GdScript
func _input(event):
	if event is InputEventKey:
		if event.scancode == KEY_W and event.shift: #W and Shift, typical hotkey.
	
	if event is InputEvenJoypadButton:
		event.button_index = SONY_X and Input.is_joy_button_pressed(0, SONY_Y) #X and Y
```

File: tutorials/inputs/input_examples.rst (hunk 1 above)
Comment (cbscribe) [reply tutorials/inputs/input_examples.rst]: Done.

**Comment by cbscribe** [comment]
I did demonstrate the key + `event.shift` method, but note I also didn't include any joypad stuff because I don't have one handy to test with. Once this is merged, someone can add a section about that.

File: tutorials/inputs/input_examples.rst [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+
+.. tabs::
+ .. code-tab:: gdscript GDScript
+
+    func _input(event):
+        if event is InputEventKey and event.pressed:
+            if event.scancode == KEY_T:
+                print("T was pressed")
+
+.. tip:: See :ref:`@GlobalScope <class_@GlobalScope>` for a list of scancode
```
Comment (merumelu) [inline tutorials/inputs/input_examples.rst]:
```suggestion
.. tip:: See :ref:`@GlobalScope <enum_@GlobalScope_KeyList>` for a list of scancode
```
You can link to the scancode enum directly.

File: tutorials/inputs/input_examples.rst [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+
+    func _input(event):
+        if event is InputEventKey and event.pressed:
+            if event.scancode == KEY_T:
+                if event.shift:
+                    print("Shift+T was pressed")
+                else:
+                    print("T was pressed")
+
+.. tip:: See :ref:`@GlobalScope <class_@GlobalScope>` for a list of scancode
```
Comment (merumelu) [inline tutorials/inputs/input_examples.rst]:
```suggestion
.. tip:: See :ref:`@GlobalScope <enum_@GlobalScope_KeyList>` for a list of scancode
```

File: tutorials/inputs/input_examples.rst [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+
+Mouse events stem from the :ref:`InputEventMouse <class_InputEventMouse>` class, and
+are separated into two types: :ref:`InputEventMouseButton <class_InputEventMouseButton>`
+and :ref:`InputEventMouseMotion <class_InputEventMouseMotion>`. Note that this
+means that all mouse events will contain a ``position`` property.
+
+Mouse buttons
+~~~~~~~~~~~~~
+
+Capturing mouse buttons is very similar to handling key events. :ref:`@GlobalScope <class_@GlobalScope>`
```
Comment (merumelu) [inline tutorials/inputs/input_examples.rst]:
```suggestion
Capturing mouse buttons is very similar to handling key events. :ref:`@GlobalScope <enum_@GlobalScope_ButtonList>`
```

Comment (cbscribe) [reply tutorials/inputs/input_examples.rst]: Good idea, I'll update those refs.

**Comment by clayjohn** [comment]
Everything looks good to me!

**Comment by cbscribe** [comment]
Thanks for the reviews.

**Pull request merged**